### PR TITLE
8740 add single processing to getting started instructions

### PR DESCRIPTION
**Pull request by pgmoka**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +108,47 @@ with the [contributors guide](https://github.com/pytorch/xla/blob/master/CONTRIB
...
+       inputs, labels = training_data[i]
++      # Transfer data to the XLA device. This happens asynchronously.
++      inputs, labels = training_data.to(inputs.xla.device()), labels.to(xla.device())
+       optimizer.zero_grad()
+       outputs = model(inputs)
+       loss = loss_fn(outputs, labels)
+       loss.backward()
+       optimizer.step()
+
++  xm.mark_step()
```
Comment (tengyifei) [inline README.md:137]: Could you replace all the `xm.mark_step()` with `torch_xla.sync()`? We're trying to encourage this simpler API

File: README.md [hunk 2]
```diff
@@ -108,6 +108,47 @@ with the [contributors guide](https://github.com/pytorch/xla/blob/master/CONTRIB
...
+Multi processing is more complex, and is not compatible with SPMD.
+This tutorial does not dive into SPMD. For more on that, check our
+[SPMD guide](https://github.com/pytorch/xla/blob/master/docs/source/perf/spmd_basic.md).
+
+### Simple single process
+
+To update your exisitng training loop, make the following changes:
+
+```diff
++import torch_xla as xla
```
Comment (tengyifei) [inline README.md:121]: nit: I've never seen anyone do `import torch_xla as xla`. I think keeping `import torch_xla` as-is will better align with the existing style.

File: README.md [hunk 3]
```diff
@@ -108,6 +108,47 @@ with the [contributors guide](https://github.com/pytorch/xla/blob/master/CONTRIB
...
++import torch_xla.core.xla_model as xm
+
+ def train(model, training_data, ...):
+   ...
+   for inputs, labels in train_loader:
++    with xla.step():
```
Comment (tengyifei) [inline README.md:127]: Did you test that this works? I've not seen this API before

File: README.md [hunk 4]
```diff
@@ -108,6 +108,47 @@ with the [contributors guide](https://github.com/pytorch/xla/blob/master/CONTRIB
...
+       inputs, labels = training_data[i]
++      # Transfer data to the XLA device. This happens asynchronously.
++      inputs, labels = training_data.to(inputs.xla.device()), labels.to(xla.device())
```
Comment (tengyifei) [inline README.md:130]: What does `inputs.xla.device()` do? I think it can be as simple as `training_data.to('xla')` here and `labels.to('xla')` next

File: README.md [hunk 5]
```diff
@@ -108,6 +108,47 @@ with the [contributors guide](https://github.com/pytorch/xla/blob/master/CONTRIB
...
+       loss.backward()
+       optimizer.step()
+
++  xm.mark_step()
+   ...
+
+ if __name__ == '__main__':
+   ...
++  # Move the model paramters to your XLA device
++  model.to(xla.device())
```
Comment (tengyifei) [inline README.md:143]: nit: `model.to('xla')`

I think this works as along as we `import torch_xla` at the start

File: README.md [hunk 6]
```diff
@@ -108,6 +108,47 @@ with the [contributors guide](https://github.com/pytorch/xla/blob/master/CONTRIB
 
 ## Getting Started
 
+Following here are guides for a simple single process, and multi process application.
```
Comment (tengyifei) [inline README.md:111]: Can we define the meaning of each mode:

- Single process: one Python interpreter controlling a single GPU/TPU
- Multi process: N Python interpreters are launched, corresponding to N GPU/TPUs found on the system
- SPMD: one Python interpreter controls all N GPU/TPUs found on the system

**Comment by tengyifei** [comment]
Changes requested

File: README.md (hunk 4 above)
Comment (pgmoka) [reply README.md]: Also, this catches a typo here. It should be:
```
inputs, labels = inputs.to('xla'), labels.to('xla')
```

File: README.md (hunk 6 above)
Comment (pgmoka) [reply README.md]: I agree these descriptions for the first page guide are useful. I have added the descriptions.

File: README.md (hunk 3 above)
Comment (tengyifei) [reply README.md]: Ah! Seems fine then!

File: README.md [hunk 7]
```diff
@@ -108,23 +108,67 @@ with the [contributors guide](https://github.com/pytorch/xla/blob/master/CONTRIB
 
 ## Getting Started
 
+Following here are guides for two modes:
+- Single process: one Python interpreter controlling a single GPU/TPU at a time
+- Multi process: N Python interpreters are launched, corresponding to N GPU/TPUs
+found on the system
+
+Relevant to the different modes is SPMD, a process in which one Python interpreter
```
Comment (tengyifei) [inline README.md:116]: nit: SPMD is not implemented as a "process" in PyTorch/XLA. It's literally yet another mode where you enter SPMD mode by calling `torch_xla.runtime.use_spmd()`. Maybe rephrase as "Another mode is SPMD, where ..."

**Review by tengyifei — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (pgmoka) [reply README.md]: I will keep only this change for now.

File: README.md (hunk 7 above)
Comment (pgmoka) [reply README.md]: That is a good phrasing check. Added suggestion.
